src/librustc/ty/constness.rs

```diff
@@ -63,13 +64,63 @@ impl<'tcx> TyCtxt<'tcx> {
 
 
 pub fn provide(providers: &mut Providers<'_>) {
-    /// only checks whether the function has a `const` modifier
+    /// Const evaluability whitelist is here to check evaluability at the
+    /// top level beforehand.
+    fn is_const_intrinsic(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
+        match tcx.fn_sig(def_id).abi() {
+            Abi::RustIntrinsic |
+            Abi::PlatformIntrinsic => {
+                match tcx.item_name(def_id) {
```

Comment:
`pref_align_of` and `ptr_offset_from` are missing from this list.

Could you add comments here and in `interpret/intrinsics.rs` saying that these lists need to be kept in sync? Also it would help if you could sort this list the same way  `interpret/intrinsics.rs` is sorted, to make it easier to match them up.

Reply:
I added them and ordered the list the same and left comments stating that these should be kept in sync.